test/features/deploy_kogito_runtime.feature

```diff
@@ -152,4 +198,148 @@ Feature: Deploy Kogito Runtime
+
+#####
+
+  @persistence
+  @infinispan
+  Scenario: Deploy <example-service> service with Maven profile <profile> using external Infinispan
+    Given Kogito Operator is deployed with Infinispan operator
+    And Infinispan instance "external-infinispan" is deployed with configuration:
+      | username | developer |
+      | password | mypass    |
+    And Clone Kogito examples into local directory
+    And Local example service "<example-service>" is built by Maven using profile "<profile>" and deployed to runtime registry
+    When Deploy <runtime> example service "<example-service>" from runtime registry with configuration:
+      | config     | enablePersistence | enabled                   |
+      | infinispan | username          | developer                 |
+      | infinispan | password          | mypass                    |
+      | infinispan | uri               | external-infinispan:11222 |
+    And Kogito Runtime "<example-service>" has 1 pods running within 10 minutes
+    And Start "orders" process on service "<example-service>" within 3 minutes with body:
+      """json
+      {
+        "approver" : "john",
+        "order" : {
+          "orderNumber" : "12345",
+          "shipped" : false
+        }
+      }
+      """
+    And Service "<example-service>" contains 1 instances of process with name "orders"
+
+    When Scale Kogito Runtime "<example-service>" to 0 pods within 2 minutes
+    And Scale Kogito Runtime "<example-service>" to 1 pods within 2 minutes
+
+    Then Service "<example-service>" contains 1 instances of process with name "orders" within 2 minutes
+
+    @springboot
+    Examples:
+      | runtime    | example-service            | profile     |
+      | springboot | process-springboot-example | persistence |
+
+    @quarkus
+    Examples:
+      | runtime    | example-service         | profile     |
+      | quarkus    | process-quarkus-example | persistence |
+
+    @quarkus
+    @native
+    Examples:
+      | runtime    | example-service         | profile            |
+      | quarkus    | process-quarkus-example | native,persistence |
+
+#####
+
+  @events
+  @kafka
+  @infinispan
+  Scenario: Deploy <example-service> with Maven profile <profile> with events using external Kafka
+    Given Kogito Operator is deployed with Infinispan and Kafka operators
+    And Kafka instance "external-kafka" is deployed
+    And Install Kogito Data Index with 1 replicas with configuration:
+      | kafka | externalURI | external-kafka-kafka-bootstrap:9092 |
+    And Clone Kogito examples into local directory
+    And Local example service "<example-service>" is built by Maven using profile "<profile>" and deployed to runtime registry
+    When Deploy <runtime> example service "<example-service>" from runtime registry with configuration:
+      | config | enableEvents      | enabled                             |
+      | config | enablePersistence | enabled                             |
```

Comment:
Well, actually no.
This is related to (mis)compatibility between Kogito CLI and CR content. To cover both options I need to distinguish several cases:
- No persistence/events - nothing to be specified in config
- Persistence/events using KogitoInfra:
    - KogitoRuntime or JobsService
         - CLI - need to apply `--enable-persistence` or `--enable-events` (indicates that I want persistence but I don't want external persistence => KogitoInfra is used)
         - CR - need to set `useKogitoInfra` in CR, done by `| infinispan | useKogitoInfra | enabled |`
    - DataIndex
         - CLI - nothing is specified (`--enable-persistence` and `--enable-events` don't exist for DataIndex)
         - CR - need to set `useKogitoInfra` in CR, done by `| infinispan | useKogitoInfra | enabled |`
- Persistence/events using external component:
    - KogitoRuntime or JobsService
         - CLI - need to apply `--enable-persistence` or `--enable-events` (indicates that I want persistence) and also connection properties, for example for `--infinispan-url` I need to specify `| infinispan | uri               | external-infinispan:11222 |`
         - CR - need to set just Infinispan URL by `| infinispan | uri               | external-infinispan:11222 |`
    - DataIndex
         - CLI - only `--infinispan-url` is neded, done by `| infinispan | uri               | external-infinispan:11222 |`
         - CR - need to set just Infinispan URL by `| infinispan | uri               | external-infinispan:11222 |`

To cover those combination with shared code between all Kogito applications and to avoid keeping messy logic in the code I decided to have separate parameters for CR and CLI. 
If you want persistence/events with KogitoInfra then you need to define `| config | enablePersistence | enabled    |` (for CLI except of DataIndex) and `| infinispan | useKogitoInfra    | enabled |` for CR. 
If you want persistence/events by external component then you need to define `| config | enablePersistence | enabled    |` (for CLI except of DataIndex) and connection properties (for both CLI and CR). 

I am open for improvements/suggestions.

Reply:
It would be simpler to only have one parameter:

```
| config | enablePersistence | enabled |
```

And remove the useKogitoInfra parameters. Therefore, if nothing else is provided, internally we enable the infinispan kogito infra. Otherwise, if the infinispan properties are provided, we just populate them. 

Maybe I didn't see that some of the above combinations are not being covered.

Reply:
hmm, that could work